File: _BHCVRP_Python_Version/cujae_inf_citi_om/exceptions/TimeWindowException.py
class TimeWindowException(Exception):
    def __init__(self, time_window=None):
        if time_window is not None:
            self.initial_node = time_window
            self.end_node = time_window

    def __init__(self, message=None):
        if message is not None:
            super().__init__(message)

    def get_initial_node(self):
        return getattr(self, "time_window", None)

    def get_end_node(self):
        return getattr(self, "time_window", None)

File: BHCVRP_Java_Version/src/cujae/inf/citi/om/exceptions/TimeWindowException.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package cujae.inf.citi.om.exceptions;

/**
 *
 * @author kmych
 */
public class TimeWindowException extends Exception{
    private float initialNode;
    private float endNode;
    
    public TimeWindowException(float initialNode, float endNode){
        this.initialNode = initialNode;
        this.endNode = endNode;
    
    }
    
    public float getInitialNode(){
        return initialNode;
    }
    
    public float getEndNode(){
        return endNode;
    }
    
    public TimeWindowException(String message){   
        super(message);
    }
    
}
